apps/_dashboard/__init__.py

```diff
@@ -135,10 +201,116 @@ def logout():
 
-    @action("dbadmin")
+    @action("dbadmin/<app>/<dbname>/<tablename>", method=["GET", "POST"])
+    @action("dbadmin/<app>/<dbname>/<tablename>/<path:path>", method=["GET", "POST"])
     @action.uses(Logged(session), "dbadmin.html")
-    def dbadmin():
-        return dict(languages=dumps(getattr(T.local, "language", {})))
+    def dbadmin(app, dbname, tablename, path=None):
+        if MODE != "full":
+            raise HTTP(403)
+
+        module = Reloader.MODULES[app]
+
+        databases = [
+            name for name in dir(module) if isinstance(getattr(module, name), DAL)
+        ]
+        if dbname not in databases:
+            raise HTTP(406)
+
+        db = getattr(module, dbname)
+
+        grid_param = dict(
+            rows_per_page=20,
+            include_action_button_text=True,
+            search_button_text=None,
+            formstyle=FormStyleFuture,
+            grid_class_style=GridClassStyleFuture,
+            auto_process=False,
+        )
+
+        if tablename not in db:
+            raise HTTP(406)
+
+        table = getattr(db, tablename)
+
+        for field in table:
+            field.readable = True
+            field.writable = True
+
+        query = table.id > 0
+        orderby = [table.id]
+        columns = [field for field in table]
+        columns = columns[:6]
+
+        def search_by_field(field):
+            if field.type == "string":
+                query = lambda value: field.startswith(value)
+            if field.type == "boolean":
+                query = lambda value: field == to_bool(value)
+            if field.type.startswith("list:"):
+                query = lambda value: field.contains(value)
+            else:
+                query = lambda value: field == value
+            return query
+
+        search_queries = [["Smart", smart_search(table)]]
+        search_queries += [
+            [f"{field.label}", search_by_field(field)]
+            for field in table
+            if field.type.split()[0]
+            in (
+                "string",
+                "integer",
+                "boolean",
+                "reference",
+                "list:string",
+                "list:integer",
+                "list:reference",
+            )
+        ]
+
+        grid = Grid(
+            path,
+            query,
+            columns=columns,
+            search_queries=search_queries,
+            orderby=orderby,
+            show_id=True,
+            T=T,
+            **grid_param,
+        )
+
+        grid.param.new_sidecar = A("Cancel")
+        grid.param.edit_sidecar = A("Cancel")
+
+        grid.formatters_by_type["date"] = (
+            lambda value: value.strftime("%m/%d/%Y") if value else ""
+        )
+
+        grid.formatters_by_type["time"] = (
+            lambda value: value.strftime("%H:%M:%S") if value else ""
+        )
+
+        grid.formatters_by_type["datetime"] = (
+            lambda value: value.isoformat() if value else ""
+        )
```

Comment:
These formatters were included to replace the javascript timezone based formatting that already exists in the Grid due to HTMX's issues with injecting Javascript. If we are going back to just normal HTML then I do believe that these are unecessary but I haven't tried it.